prov/rxd/src/rxd_av.c

```diff
@@ -251,35 +231,49 @@ static int rxd_av_remove(struct fid_av *av_fid, fi_addr_t *fi_addr, size_t count
 			uint64_t flags)
 {
 	int ret = 0;
-	size_t i, addrlen;
+	size_t i;
 	fi_addr_t rxd_addr;
 	struct rxd_av *av;
-	uint8_t addr[RXD_NAME_LENGTH];
+    struct dlist_entry *av_entry;
+    struct util_ep *util_ep;
+    struct rxd_ep *rxd_ep;
+    struct rxd_peer *peer_entry;
+    struct rxd_fiaddr_entry *entry;
 
 	av = container_of(av_fid, struct rxd_av, util_av.av_fid);
 	fastlock_acquire(&av->util_av.lock);
 	for (i = 0; i < count; i++) {
-		rxd_addr = av->fi_addr_table[fi_addr[i]];
 
-		addrlen = RXD_NAME_LENGTH;
-		ret = fi_av_lookup(av->dg_av, av->rxd_addr_table[rxd_addr].dg_addr,
-				   addr, &addrlen);
-		if (ret)
-			goto err;
-		
-		ret = ofi_rbmap_find_delete(&av->rbmap, (void *) addr);
-		if (ret)
-			goto err;
-
-		ret = fi_av_remove(av->dg_av, &av->rxd_addr_table[rxd_addr].dg_addr,
-				   1, flags);
-		if (ret)
-			goto err;
-
-		av->fi_addr_table[fi_addr[i]] = FI_ADDR_UNSPEC;
-		av->rxd_addr_table[rxd_addr].fi_addr = FI_ADDR_UNSPEC;
-		av->rxd_addr_table[rxd_addr].dg_addr = FI_ADDR_UNSPEC;
-		av->dg_av_used--;
+        // remove mappings between fi_addr --> dg_addr
+        //  remove each endpoint hash mapping between 
+        // fi_addr --> rxd_addr.
+        // reset peer_entry->fi_addr value to UNSPEC
+        // do not delete the peer entry. it may be being used 
+        // for an underlying existing communication. 
+        // donot delete dg_addr from dg_av.
+        ret =  ofi_av_remove_addr(&av->util_av, fi_addr[i]);
+        if (ret)
+            goto err;
+
+        dlist_foreach(&av->util_av.ep_list, av_entry) {
+
+            HASH_FIND(hh, rxd_ep->fi_rxdaddr_hash, (void*)(&fi_addr[i]), 
+                        sizeof(fi_addr_t), entry);
```

Comment:
Whenever you're using sizeof() try to use sizeof(var) instead of sizeof(type) if possible to make it easier to maintain. ie `sizeof(fi_addr[i])` instead of `sizeof(fi_addr_t)`